prelude/slang-cpp-types.h

```diff
@@ -232,6 +232,10 @@ struct ComputeVaryingInput
     uint3 groupThreadID;
 };
 
+/* Type that defines the uniform entry point params. The actual content of this type is dependent on the entry point parameters, and can be
+found via reflection or defined such that it matches the shader appropriately. */
+struct UniformEntryPointParams;
```

Comment:
Could this lead to "one definition rule" violations if we ever compiled multiple entry points (each with their own `UniformEntryPointParams` into a single binary).

I'm not really *that* concerned about this, but it seems like it would actually be safer in general to only forward-declare your entry point functions with either a `void*` for the entry-point params or a one-off `struct` type declared just for a specific entry point. Trying to have a catch-all type here implies that they are all the same, no?